.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,125 @@
+name: "Integration tests"
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+
+jobs:
+  integration-tests:
+    runs-on: ubuntu-latest
+    steps:
+      # Need to check out repo so that 'gh pr' will know the correct repo
+      - uses: actions/checkout@v3
+
+      - name: Trigger variant analysis
+        id: trigger
+        run: |
+          if [ "$GITHUB_EVENT_NAME" == "pull_request" ]; then
+            REF="$GITHUB_HEAD_REF"
+          else
+            REF="$GITHUB_REF_NAME"
+          fi
+
+          QUERY_PACK=$(curl https://github.com/$GITHUB_REPOSITORY/releases/download/test/test_pack2.tar.gz -L | base64)
+
+          cat <<EOF >> input.json
+            {
+              "action_repo_ref": "$REF",
+              "language": "go",
+              "query_pack": "$QUERY_PACK",
+              "repositories": [
+                "docker/compose",
+                "hashicorp/terraform",
+                "github/does-not-exist"
+              ]
+            }
+          EOF
+          echo "input.json: $(cat input.json)"
+
+          RESPONSE=$(curl -H "Authorization: Bearer ${{ secrets.BOT_TOKEN }}" "https://api.github.com/repos/$GITHUB_REPOSITORY/code-scanning/codeql/variant-analyses" -X POST -d @input.json)
+          echo "Response: $RESPONSE"
+
+          ID="$(echo "$RESPONSE" | jq '.id')"
+          echo "Triggered variant analysis $ID"
+
+          if [ "$ID" == "null" ]; then
+            echo "Error triggering variant analysis"
+            exit 1
+          fi
+
+          echo "::set-output name=variant_anlaysis_id::$ID"
+
+      - name: Wait for variant analysis to complete
+        run: |
+          while true; do
+            RESPONSE=$(curl -H "Authorization: Bearer ${{ secrets.BOT_TOKEN }}" "https://api.github.com/repos/$GITHUB_REPOSITORY/code-scanning/codeql/variant-analyses/${{ steps.trigger.outputs.variant_anlaysis_id }}")
+            STATUS="$(echo "$RESPONSE" | jq '.status' -r)"
+            echo "Variant analysis ${{ steps.trigger.outputs.variant_anlaysis_id }} status: $STATUS"
+            if [ "$STATUS" == "completed" ]; then
+              echo "Exiting..."
+              exit 0
+            fi
+            sleep 10s
```

Comment:
Should we put a timeout on this workflow of e.g. 30 minutes? Otherwise we would keep polling for 6 hours.

Reply:
Good point. A timeout on the workflow would stop us wasting compute time, and if the variant analysis takes longer than (e.g.) 30 minutes then something is wrong and we can consider it to have failed.